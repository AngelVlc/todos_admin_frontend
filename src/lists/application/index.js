export * from "./GetListByIdUseCase";
export * from "./DeleteListByIdUseCase";
export * from "./GetListsUseCase";
export * from "./CreateListUseCase";
export * from "./UpdateListUseCase";
export * from "./MoveListItemUseCase";
export * from "./IndexAllListsUseCase";
export * from "./GetSearchSecureKeyUseCase";
