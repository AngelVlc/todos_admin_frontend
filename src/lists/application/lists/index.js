export * from "./GetListByIdUseCase";
export * from "./DeleteListByIdUseCase";
