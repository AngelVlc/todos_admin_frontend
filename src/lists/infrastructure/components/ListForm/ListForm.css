.dnd-item {
  border: 1px solid lightgrey;
  border-radius: 2px;
}

.rounded-button {
  background-color: rgba(10, 10, 10, 0.2);
  border: none;
  border-radius: 9999px;
  color: white;
  height: 20px;
  width: 20px;
  vertical-align: top;
}
