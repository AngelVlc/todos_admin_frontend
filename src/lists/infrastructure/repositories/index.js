export * from "./ListsRepository";
export * from "./ListItemsRepository";
