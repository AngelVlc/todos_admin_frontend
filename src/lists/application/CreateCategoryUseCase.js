import { BaseUseCase } from "../../shared/domain/BaseUseCase";
import { CategoriesRepository } from "../infrastructure/repositories";

export class CreateCategoryUseCase extends BaseUseCase {
  _categoriesRepository;

  constructor({ categoriesRepository }) {
    super();

    this._categoriesRepository = categoriesRepository;
  }

  static create() {
    const categoriesRepository = new CategoriesRepository();

    return new CreateCategoryUseCase({ categoriesRepository });
  }

  async execute(category) {
   return await this._categoriesRepository.create(category);
  }
}
