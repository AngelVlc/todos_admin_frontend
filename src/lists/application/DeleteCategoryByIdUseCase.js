import { BaseUseCase } from "../../shared/domain/BaseUseCase";
import { CategoriesRepository } from "../infrastructure/repositories";

export class DeleteCategoryByIdUseCase extends BaseUseCase {
  _repository;

  constructor({ repository }) {
    super();

    this._repository = repository;
  }

  static create() {
    const repository = new CategoriesRepository();

    return new DeleteCategoryByIdUseCase({ repository });
  }

  async execute(categoryId) {
    return await this._repository.deleteById(categoryId);
  }
}
