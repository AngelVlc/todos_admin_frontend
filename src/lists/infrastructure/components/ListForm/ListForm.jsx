import React, { useState, useContext, useRef } from "react";
import { useHistory } from "react-router-dom";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
import * as Yup from "yup";
import "./ListForm.css";
import { List, ListItem } from "../../../domain";
import {
  CreateListUseCase,
  UpdateListUseCase,
} from "../../../application/lists";
import { Modal } from "../../../../shared/infrastructure/components/Modal";
import { ListItemForm } from "../ListItemForm";

export const ListForm = (props) => {
  let history = useHistory();
  const { useCaseFactory } = useContext(AppContext);
  const itemFormModalRef = useRef();
  const itemFormRef = useRef();

  const [pageState, setPageState] = useState(props.list);

  const onSubmit = async (list) => {
    let useCase;

    if (props.list?.id === -1) {
      useCase = useCaseFactory.get(CreateListUseCase);
    } else {
      useCase = useCaseFactory.get(UpdateListUseCase);
    }

    const result = await useCase.execute(list);

    if (result && props.list?.id === -1) {
      history.push(`/lists/${result.id}`);
    }
  };

  const onDragEnd = (dropResult) => {
    const { destination, source } = dropResult;

    if (!destination) {
      return;
    }

    if (
      destination.droppableId === source.droppableId &&
      destination.index === source.index
    ) {
      return;
    }

    const newItems = Array.from(pageState.items);
    newItems.splice(source.index, 1);
    newItems.splice(destination.index, 0, pageState.items[source.index]);

    const newList = new List({ ...pageState, items: newItems });

    setPageState(newList);
  };

  const onDeleteListItem = (item) => {
    const index = pageState.items.indexOf(item);
    const newItems = Array.from(pageState.items);
    newItems[index].state = "deleted";

    const newList = new List({ ...pageState, items: newItems });

    setPageState(newList);
  };

  const onAddNewItem = () => {
    showItemModal(ListItem.createEmpty(props.list?.id));
  };

  const onEditItem = (item) => {
    showItemModal(item);
  };

  const showItemModal = (item) => {
    itemFormRef.current.setValues(item);
    itemFormModalRef.current.showModal();
  };

  const onSubmitItemForm = (listItem) => {
    const newList = new List({ ...pageState, items: pageState.items });

    if (listItem.id === -1) {
      newList.addNewItem(listItem);
    } else {
      for (const item of newList.items) {
        if (item.id === listItem.id) {
          item.title = listItem.title;
          item.description = listItem.description;
          item.state = "modified";
          item.listId = listItem.listId;

          break;
        }
      }
    }

    setPageState(newList);

    itemFormModalRef.current.closeModal();
  };

  const onItemModalClose = () => {
    itemFormRef.current.submitForm();
  };

  return (
    <>
      <Modal ref={itemFormModalRef} closeHandler={onItemModalClose}>
        <ListItemForm ref={itemFormRef} onSubmit={onSubmitItemForm} />
      </Modal>
      <Formik
        enableReinitialize={true}
        initialValues={pageState}
        validationSchema={Yup.object({
          name: Yup.string().required("Required"),
        })}
        onSubmit={onSubmit}
      >
        <Form>
          <div className="field">
            <label className="label" htmlFor="name">
              Name
            </label>
            <div className="control">
              <Field
                name="name"
                as="input"
                className="input"
                data-testid="name"
                autoFocus
              />
            </div>
            <p className="help is-danger" data-testid="nameErrors">
              <ErrorMessage name="name" />
            </p>
          </div>
          <div>
            <div className="is-flex">
              <div className="is-flex-grow-4">
                <span className="label">List Items</span>
              </div>
            </div>
            <DragDropContext onDragEnd={onDragEnd}>
              <div className="dnd-container">
                <div className="list-button mt-2 mb-2">
                  <button
                    className="button is-small"
                    type="button"
                    data-testid="addNew"
                    onClick={() => onAddNewItem()}
                  >
                    <span className="icon is-small">
                      <i className="fas fa-plus"></i>
                    </span>
                    <span>Add</span>
                  </button>
                </div>
                <Droppable droppableId={props.list.id.toString()}>
                  {(provided) => (
                    <div
                      className="dnd-list"
                      ref={provided.innerRef}
                      {...provided.droppableProps}
                    >
                      {pageState.items.length > 0 &&
                        pageState.items
                          .filter((item) => item.state !== "deleted")
                          .map((item, index) => (
                            <Draggable
                              key={item.id}
                              draggableId={item.id.toString()}
                              index={index}
                            >
                              {(draggableProvided) => (
                                <div
                                  key={item.id}
                                  className="is-flex dnd-item p-2 mb-2"
                                  {...draggableProvided.draggableProps}
                                  {...draggableProvided.dragHandleProps}
                                  ref={draggableProvided.innerRef}
                                  data-testid={`draggable${item.id}`}
                                >
                                  <div
                                    className="is-flex-grow-4"
                                    data-testid={`editListItem${item.id}`}
                                    onClick={() => onEditItem(item)}
                                  >
                                    <span className="has-text-black">
                                      {item.title}
                                    </span>
                                  </div>
                                  <div className="is-justify-content-flex-end">
                                    <center>
                                      <button
                                        type="button"
                                        className="has-text-black delete"
                                        data-testid={`deleteListItem${item.id}`}
                                        onClick={() => onDeleteListItem(item)}
                                      ></button>
                                    </center>
                                  </div>
                                </div>
                              )}
                            </Draggable>
                          ))}
                      {provided.placeholder}
                    </div>
                  )}
                </Droppable>
              </div>
            </DragDropContext>
          </div>
          <div className="field is-grouped">
            <div className="control">
              <button className="button" data-testid="submit" type="submit">
                {props.list?.id ? "SAVE" : "CREATE"}
              </button>
            </div>
            <>{props.preCancel}</>
            <div className="control ml-auto is-pulled-right">
              <button
                className="button"
                data-testid="cancel"
                type="button"
                onClick={() => history.push("/lists")}
              >
                CANCEL
              </button>
            </div>
          </div>
        </Form>
      </Formik>
    </>
  );
};
