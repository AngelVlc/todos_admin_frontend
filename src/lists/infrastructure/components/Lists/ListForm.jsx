import React, { useState, useContext } from "react";
import { useHistory, Link } from "react-router-dom";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
import * as Yup from "yup";
import "./ListForm.css";
import { List } from "../../../domain";
import {
  CreateListUseCase,
  UpdateListUseCase,
} from "../../../application/lists";

export const ListForm = (props) => {
  let history = useHistory();
  const { useCaseFactory } = useContext(AppContext);

  const [pageState, setPageState] = useState(props.list);

  const onSubmit = async (list) => {
    let useCase;
    if (props.list?.id === undefined) {
      useCase = useCaseFactory.get(CreateListUseCase);
    } else {
      useCase = useCaseFactory.get(UpdateListUseCase);
    }

    const result = await useCase.execute(list);
    if (result && props.list?.id === undefined) {
      history.push(`/lists/${result.id}`);
    }
  };

  const onDragEnd = (dropResult) => {
    const { destination, source } = dropResult;

    if (!destination) {
      return;
    }

    if (
      destination.droppableId === source.droppableId &&
      destination.index === source.index
    ) {
      return;
    }

    const newItems = Array.from(pageState.items);
    newItems.splice(source.index, 1);
    newItems.splice(destination.index, 0, pageState.items[source.index]);

    setPageState(new List({ ...pageState, items: newItems }));
  };

  return (
    <Formik
      enableReinitialize={true}
      initialValues={pageState}
      validationSchema={Yup.object({
        name: Yup.string().required("Required"),
      })}
      onSubmit={onSubmit}
    >
      <Form>
        <div className="field">
          <label className="label" htmlFor="name">
            Name
          </label>
          <div className="control">
            <Field
              name="name"
              as="input"
              className="input"
              data-testid="name"
              autoFocus
            />
          </div>
          <p className="help is-danger" data-testid="nameErrors">
            <ErrorMessage name="name" />
          </p>
        </div>
        <div className="field">
          <label className="label" htmlFor="isQuickList">
            Is Quick List
          </label>
          <div className="control">
            <Field
              name="isQuickList"
              type="checkbox"
              data-testid="isQuickList"
            />
          </div>
        </div>
        {props.list?.id && (
          <div>
            <div className="is-flex">
              <div className="is-flex-grow-4">
                <span className="label">List Items</span>
              </div>
            </div>
            <DragDropContext onDragEnd={onDragEnd}>
              <div className="dnd-container">
                <div className="list-button mt-2 mb-2">
                  <button
                    className="button is-small"
                    type="button"
                    data-testid="addNew"
                    onClick={() =>
                      history.push(`/lists/${props.list.id}/items/new`)
                    }
                  >
                    <span className="icon is-small">
                      <i className="fas fa-plus"></i>
                    </span>
                    <span>Add</span>
                  </button>
                </div>
                <Droppable droppableId={props.list.id.toString()}>
                  {(provided) => (
                    <div
                      className="dnd-list"
                      ref={provided.innerRef}
                      {...provided.droppableProps}
                    >
                      {pageState.items.length > 0 &&
                        pageState.items.map((item, index) => (
                          <Draggable
                            key={item.id}
                            draggableId={item.id.toString()}
                            index={index}
                          >
                            {(draggableProvided) => (
                              <div
                                key={item.id}
                                className="is-flex dnd-item p-2 mb-2"
                                {...draggableProvided.draggableProps}
                                {...draggableProvided.dragHandleProps}
                                ref={draggableProvided.innerRef}
                                data-testid={`draggable${item.id}`}
                              >
                                <Link
                                  className="is-flex-grow-4"
                                  data-testid={`editListItem${item.id}`}
                                  to={`/lists/${props.list.id}/items/${item.id}/edit`}
                                >
                                  <div>
                                    <span className="has-text-black">
                                      {item.title}
                                    </span>
                                  </div>
                                </Link>
                                <div className="is-justify-content-flex-end">
                                  <center>
                                    <Link
                                      className="has-text-black delete"
                                      data-testid={`deleteListItem${item.id}`}
                                      to={`/lists/${props.list.id}/items/${item.id}/delete`}
                                    ></Link>
                                  </center>
                                </div>
                              </div>
                            )}
                          </Draggable>
                        ))}
                      {provided.placeholder}
                    </div>
                  )}
                </Droppable>
              </div>
            </DragDropContext>
          </div>
        )}

        <div className="field is-grouped">
          <div className="control">
            <button className="button" data-testid="submit" type="submit">
              {props.list?.id ? "SAVE" : "CREATE"}
            </button>
          </div>
          {props.list?.id && (
            <div className="control">
              <button
                className="button"
                data-testid="read"
                type="button"
                onClick={() => history.push(`/lists/${props.list.id}/read`)}
              >
                READ
              </button>
            </div>
          )}
          <div className="control">
            <button
              className="button"
              data-testid="cancel"
              type="button"
              onClick={() => history.push("/lists")}
            >
              CANCEL
            </button>
          </div>
          {props.list?.id && (
            <div className="control ml-auto is-pulled-right">
              <button
                className="button is-danger"
                data-testid="delete"
                type="button"
                onClick={() => history.push(`/lists/${props.list.id}/delete`)}
              >
                DELETE
              </button>
            </div>
          )}
        </div>
      </Form>
    </Formik>
  );
};
