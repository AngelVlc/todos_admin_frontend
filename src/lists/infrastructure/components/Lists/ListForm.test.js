import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { ListForm } from "./ListForm";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { MemoryRouter, Route } from "react-router-dom";
import { act } from "react-dom/test-utils";
import {
  mockGetComputedStyle,
  mockDndSpacing,
  makeDnd,
  DND_DIRECTION_DOWN,
} from "react-beautiful-dnd-test-utils";
import { CreateListUseCase } from "../../../application/lists";
import { List, ListItem } from "../../../domain";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const mockedCreateListUseCase = {
  execute: jest.fn(),
};

const mockedUpdateListUseCase = {
  execute: jest.fn(),
};

const useCaseFactory = {
  get: (useCase) =>
    useCase == CreateListUseCase
      ? mockedCreateListUseCase
      : mockedUpdateListUseCase,
};

beforeEach(() => {
  mockGetComputedStyle();
});

afterEach(cleanup);

describe("ListForm", () => {
  describe("when the list already exist", () => {
    const renderWithContextAndRouterForExistingList = () => {
      const list = new List({ id: 2, name: "list name" });
      list.items = [
        new ListItem({
          id: 5,
          listId: 2,
          title: "item 5 title",
          description: "item 5 description",
        }),
        new ListItem({
          id: 6,
          listId: 2,
          title: "item 6 title",
          description: "item 6 description",
        }),
      ];

      const context = { auth: { info: {} }, useCaseFactory };
      return {
        ...render(
          <AppContext.Provider value={context}>
            <MemoryRouter initialEntries={[`/lists/2/items/5/edit`]}>
              <Route path="/lists/:listId/items/:itemId/edit">
                <ListForm list={list} />
              </Route>
            </MemoryRouter>
          </AppContext.Provider>
        ),
      };
    };

    it("should match the snapshot", async () => {
      const { asFragment } = renderWithContextAndRouterForExistingList();

      expect(asFragment()).toMatchSnapshot();
    });

    it("should allow to update the list name, delete and update an existing item, and add a new item", async () => {
      const container = renderWithContextAndRouterForExistingList();

      mockDndSpacing(container.container);

      await makeDnd({
        getDragElement: () => container.getByTestId("draggable5"),
        direction: DND_DIRECTION_DOWN,
        positions: 1,
      });

      await waitFor(() => {
        fireEvent.click(container.getByTestId("deleteListItem5"));
      });

      await waitFor(() => {
        fireEvent.click(container.getByTestId("addNew"));
      });

      await changeInputValue(container.getByTestId, "title", "the title");
      await changeInputValue(
        container.getByTestId,
        "description",
        "the description"
      );

      await waitFor(() => {
        fireEvent.click(container.getByTestId("modalOk"));
      });

      await waitFor(() => {
        fireEvent.click(container.getByTestId("editListItem6"));
      });

      await changeInputValue(container.getByTestId, "title", "updated title");
      await changeInputValue(
        container.getByTestId,
        "description",
        "updated description"
      );

      await waitFor(() => {
        fireEvent.click(container.getByTestId("modalOk"));
      });

      await changeInputValue(container.getByTestId, "name", "updated name");

      mockedUpdateListUseCase.execute.mockResolvedValue({ id: 2 });

      await waitFor(() => {
        fireEvent.click(container.getByTestId("submit"));
      });

      expect(mockedUpdateListUseCase.execute).toHaveBeenCalled();
      const items = [
        new ListItem({
          id: 6,
          listId: 2,
          title: "updated title",
          description: "updated description",
          state: "modified",
        }),
        new ListItem({
          id: 5,
          listId: 2,
          title: "item 5 title",
          description: "item 5 description",
          state: "deleted",
        }),
        new ListItem({
          id: -1,
          listId: 2,
          title: "the title",
          description: "the description",
          position: 2,
        }),
      ];
      const list = new List({
        id: 2,
        name: "updated name",
      });
      list.items = items;

      expect(mockedUpdateListUseCase.execute.mock.calls[0][0]).toStrictEqual(
        list
      );
    });

    it("should allow to delete an existing item", async () => {
      const { getByTestId } = renderWithContextAndRouterForExistingList();

      await waitFor(() => {
        fireEvent.click(getByTestId("deleteListItem5"));
      });

      mockedUpdateListUseCase.execute.mockResolvedValue({ id: 2 });

      await waitFor(() => {
        fireEvent.click(getByTestId("submit"));
      });

      expect(mockedUpdateListUseCase.execute).toHaveBeenCalled();

      const items = [
        new ListItem({
          id: 5,
          title: "item 5 title",
          description: "item 5 description",
          state: "deleted",
          listId: 2,
        }),
        new ListItem({
          id: 6,
          title: "item 6 title",
          description: "item 6 description",
          listId: 2,
        }),
      ];

      const list = new List({
        id: 2,
        name: "list name",
      });
      list.items = items;

      expect(mockedUpdateListUseCase.execute.mock.calls[0][0]).toStrictEqual(
        list
      );
    });
  });

  describe("when the list is new", () => {
    const renderWithContextAndRouterForNewList = () => {
      const context = { auth: { info: {} }, useCaseFactory };
      return {
        ...render(
          <AppContext.Provider value={context}>
            <MemoryRouter initialEntries={[`/users/new`]}>
              <Route path="/users/new">
                <ListForm list={List.createEmpty()} />
              </Route>
            </MemoryRouter>
          </AppContext.Provider>
        ),
      };
    };

    it("should match the snapshot", async () => {
      const { asFragment } = renderWithContextAndRouterForNewList();

      expect(asFragment()).toMatchSnapshot();
    });

    it("should allow to cancel", async () => {
      const { getByTestId } = renderWithContextAndRouterForNewList();

      await waitFor(() => {
        fireEvent.click(getByTestId("cancel"));
      });

      expect(mockHistoryPush).toHaveBeenCalled();
      expect(mockHistoryPush.mock.calls[0][0]).toBe("/lists");
      mockHistoryPush.mockClear();
    });

    it("should require the list name", async () => {
      const { getByTestId } = renderWithContextAndRouterForNewList();

      await waitFor(() => {
        fireEvent.click(getByTestId("submit"));
      });

      expect(getByTestId("nameErrors")).toHaveTextContent("Required");
    });

    it("should allow to create a new list", async () => {
      const { getByTestId } = renderWithContextAndRouterForNewList();

      await changeInputValue(getByTestId, "name", "new list");

      await waitFor(() => {
        fireEvent.click(getByTestId("addNew"));
      });

      await changeInputValue(getByTestId, "title", "the title");
      await changeInputValue(getByTestId, "description", "the description");

      await waitFor(() => {
        fireEvent.click(getByTestId("modalOk"));
      });

      mockedCreateListUseCase.execute.mockResolvedValue({ id: 55 });

      await waitFor(() => {
        fireEvent.click(getByTestId("submit"));
      });

      const list = new List({
        id: -1,
        name: "new list",
        itemsCount: 0,
      });
      list.items = [
        new ListItem({
          id: -1,
          listId: -1,
          title: "the title",
          description: "the description",
          position: 0,
        }),
      ];

      expect(mockedCreateListUseCase.execute).toHaveBeenCalled();
      expect(mockedCreateListUseCase.execute.mock.calls[0][0]).toStrictEqual(
        list
      );

      expect(mockHistoryPush).toHaveBeenCalled();
      expect(mockHistoryPush.mock.calls[0][0]).toBe("/lists/55");
      mockHistoryPush.mockClear();
    });
  });
});

const changeInputValue = async (getByTestId, name, value) => {
  await waitFor(() => {
    fireEvent.change(getByTestId(name), {
      target: {
        value: value,
      },
    });
  });
};
