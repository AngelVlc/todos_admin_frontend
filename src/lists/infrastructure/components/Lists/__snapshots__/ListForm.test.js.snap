// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for a new list 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="field"
    >
      <label
        class="label"
        for="name"
      >
        Name
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          data-testid="name"
          name="name"
          value=""
        />
      </div>
      <p
        class="help is-danger"
        data-testid="nameErrors"
      />
    </div>
    <div
      class="field is-grouped"
    >
      <p
        class="control"
      >
        <button
          class="button"
          data-testid="submit"
          type="submit"
        >
          CREATE
        </button>
      </p>
      <p
        class="control"
      >
        <button
          class="button"
          data-testid="cancel"
          type="button"
        >
          CANCEL
        </button>
      </p>
    </div>
  </form>
</DocumentFragment>
`;

exports[`should match the snapshot for an existing list 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="field"
    >
      <label
        class="label"
        for="name"
      >
        Name
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          data-testid="name"
          name="name"
          value="list name"
        />
      </div>
      <p
        class="help is-danger"
        data-testid="nameErrors"
      />
    </div>
    <div>
      <div
        class="is-flex"
      >
        <div
          class="is-flex-grow-4"
        >
          <span
            class="label"
          >
            List Items
          </span>
        </div>
      </div>
      <div
        class="dnd-container"
      >
        <div
          class="list-button"
        >
          <button
            class="button is-small"
            data-testid="addNew"
            type="button"
          >
            <span
              class="icon is-small"
            >
              <i
                class="fas fa-plus"
              />
            </span>
            <span>
              Add
            </span>
          </button>
        </div>
        <div
          class="dnd-list"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="2"
        >
          <div
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            class="is-flex dnd-item"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="5"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="5"
            data-testid="draggable5"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <a
              class="is-flex-grow-4"
              data-testid="editListItem5"
              href="/lists/2/items/5/edit"
            >
              <div>
                <span
                  class="has-text-black"
                >
                  item 5 title
                </span>
              </div>
            </a>
            <div
              class="is-justify-content-flex-end"
            >
              <center>
                <a
                  class="has-text-black delete"
                  data-testid="deleteListItem5"
                  href="/lists/2/items/5/delete"
                />
              </center>
            </div>
          </div>
          <div
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            class="is-flex dnd-item"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="6"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="6"
            data-testid="draggable6"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <a
              class="is-flex-grow-4"
              data-testid="editListItem6"
              href="/lists/2/items/6/edit"
            >
              <div>
                <span
                  class="has-text-black"
                >
                  item 6 title
                </span>
              </div>
            </a>
            <div
              class="is-justify-content-flex-end"
            >
              <center>
                <a
                  class="has-text-black delete"
                  data-testid="deleteListItem6"
                  href="/lists/2/items/6/delete"
                />
              </center>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="field is-grouped"
    >
      <p
        class="control"
      >
        <button
          class="button"
          data-testid="submit"
          type="submit"
        >
          SAVE
        </button>
      </p>
      <p
        class="control"
      >
        <button
          class="button"
          data-testid="cancel"
          type="button"
        >
          CANCEL
        </button>
      </p>
      <p
        class="control"
      >
        <button
          class="button is-danger"
          data-testid="delete"
          type="button"
        >
          DELETE
        </button>
      </p>
    </div>
  </form>
</DocumentFragment>
`;
